agent/consul/server_test.go

```diff
@@ -1119,3 +1121,24 @@ func TestServer_RPC_RateLimit(t *testing.T) {
 		}
 	})
 }
+
+func TestServer_CALogging(t *testing.T) {
+	t.Parallel()
+	dir1, conf1 := testServerConfig(t)
```

Comment:
I'm not seeing any change to this config that would cause `NeedsLogger` to be set for this test?